In [2]:
import pandas as pd

df = pd.read_csv("allbacks.csv")
df.head()

books = pd.read_csv("allbacks.csv")

books

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb


In [3]:
sorted_covers = df['cover'].sort_values().reset_index(drop=True)
median_index = len(sorted_covers) // 2
print("Median Cover Type:", sorted_covers[median_index])

Median Cover Type: pb


In [5]:
mean_weight = books["weight"].mean()

print("Mean weight of all books:", mean_weight, "grams")

Mean weight of all books: 706.6666666666666 grams


In [6]:
mean_by_cover = books.groupby("cover")["weight"].mean()

print(mean_by_cover)

cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64


### Interpretation of Group Means

Hardback books have an average weight of 796.43 grams,
while paperback books have an average weight of 628.13 grams.
The difference is approximately 168.30 grams, suggesting
that hardback books in this sample tend to be heavier
than paperback books.